internal/cluster/validations/validations.go

```diff
@@ -208,8 +204,19 @@ func (v *registryPullSecretValidator) ValidatePullSecret(secret string, username
 		}
 	}
 
-	for registry := range *v.registriesWithAuth {
+	images := make([]string, len(v.images))
+	copy(images, v.images)
+
+	if releaseImageURL != "" {
+		images = append(images, releaseImageURL)
+	}
+
+	registriesWithAuth, err := getRegistriesWithAuth(v.config.PublicRegistries, ignoreListSeparator, images...)
```

Comment:
i assume it is not a big deal to go through images all the time but not sure why not to leave the logic we had + append registry to v.registriesWithAuth once you get releaseImageURL ?

Reply:
I agree with you going over the images once is better than 4 times, but I didn't want to add the release image url to that list because it is dynamic, which means we will have to delete it after. I can extract the login of `getRegistriesWithAuth` and run it once as instance creation and once again for each release image

Reply:
i assume i missed something here, can't we just copy the object and add dynamic to static when required?
or we don't have static objects anymore?

Reply:
I think what you're suggesting is to maintain a list of registries with auth based on the static images we check then still call `getRegistriesWithAuth` for the release image when it's passed. We's also still have to do the same `copy` + `append` dance but with registries rather than images.

Is that right @tsorya ?

I think it's a good idea.

Reply:
Yep :) 
Thanks @carbonin

Reply:
Done :)